api/s2n.h

```diff
@@ -596,6 +596,258 @@ extern int s2n_cert_get_utf8_string_from_extension_data_length(const uint8_t *ex
+ */
+S2N_API 
+int s2n_connection_get_negotiated_psk_identity_length(struct s2n_connection *conn, uint16_t *identity_length);
+
+/**
+ * Gets the negotiated pre-shared key identity from the s2n connection object. 
+ * If the negotiated pre-shared key does not exist, the PSK identity will not be obtained and no error will be returned. 
+ * 
+ * @param conn A pointer to the s2n_connection object.
+ * @param identity The negotiated pre-shared key identity obtained from the s2n_connection object. 
+ * @param max_identity_length The maximum length for the pre-shared key identity. If the negotiated psk_identity length is 
+ * greater than this `max_identity_length` value an error will be returned.
+ */
+S2N_API 
+int s2n_connection_get_negotiated_psk_identity(struct s2n_connection *conn, uint8_t *identity, uint16_t max_identity_length);
+
+struct s2n_offered_psk;
+
+/**
+ * Creates a new s2n offered pre-shared key object. 
+ * An offered pre-shared key object represents a single pre-shared key (PSK) shared by the client.
+ * 
+ * # Safety
+ * 
+ * Use `s2n_offered_psk_free` to free the memory allocated to the s2n offered pre-shared key object created by this API. 
+ *
+ * @return struct s2n_offered_psk* Returns a pointer to the newly created offered pre-shared key object.
+ */
+S2N_API 
+struct s2n_offered_psk* s2n_offered_psk_new();
+
+/**
+ * Frees the memory associated with the `s2n_offered_psk` object.
+ *
+ * @param psk A pointer to the `s2n_offered_psk` object to be freed.
+ */
+S2N_API 
+int s2n_offered_psk_free(struct s2n_offered_psk **psk);
+
+/**
+ * Gets the pre-shared key identity and pre-shared key identity length for a given offered pre-shared key object. 
+ * 
+ * @param psk A pointer to the offered pre-shared key object being read.
+ * @param identity The pre-shared key identity being obtained.
+ * @param size The length of the pre-shared key identity being obtained.
+ */
+S2N_API 
+int s2n_offered_psk_get_identity(struct s2n_offered_psk *psk, uint8_t** identity, uint16_t *size);
+
+struct s2n_offered_psk_list;
+
+/**
+ * Checks whether the offered pre-shared key list has an offered psk object next in line in the list.
+ * An offered pre-shared key list contains all the pre-shared keys offered by the client for the server to select.
+ *
+ * If an offered pre-shared key exists next in line, this API returns True, otherwise False. 
+ * 
+ * @param psk_list A pointer to the offered pre-shared key list being read.
+ * @return bool A boolean value representing whether an offered psk object is present next in line in the offered pre-shared key list.
+ */
+S2N_API 
+bool s2n_offered_psk_list_has_next(struct s2n_offered_psk_list *psk_list);
+
+/**
+ * Obtains the next offered pre-shared key object from the list of offered pre-shared keys. Use `s2n_offered_psk_list_has_next` 
+ * prior to this API call to determine if an offered pre-shared key exists next in the list. 
+ * 
```

Comment:
You might want to just say "call to ensure we have not reached the end of the list".